components/profile/hypercerts-tab/hypercerts-tab-content-claimable.tsx

```diff
@@ -1,19 +1,22 @@
-"use client";
-
-import ExploreListSkeleton from "@/components/explore/explore-list-skeleton";
+import { getAllowListRecordsForAddressByClaimedWithMetadata } from "@/allowlists/actions/getAllowListRecordsForAddressByClaimedWithMetadata";
 import { EmptySection } from "@/components/global/sections";
 import UnclaimedHypercertsList from "@/components/profile/unclaimed-hypercerts-list";
-import { useClaimableHypercerts } from "@/hooks/useClaimableHypercerts";
 
-export const ClaimableContent = ({ address }: { address: string }) => {
-  const { data: response, isLoading } = useClaimableHypercerts(address);
+interface ClaimableContentProps {
+  address: string;
+}
 
-  if (isLoading) return <ExploreListSkeleton length={4} />;
-  if (!response) return <EmptySection />;
+export async function ClaimableContent({ address }: ClaimableContentProps) {
+  const response = await getAllowListRecordsForAddressByClaimedWithMetadata({
+    address,
+    claimed: false,
+  });
 
-  const { data } = response;
+  if (!response || !response.data || response.data.length === 0) {
+    return <EmptySection />;
+  }
 
-  if (!data || data.length === 0) return <EmptySection />;
+  const { data } = response;
 
-  return <UnclaimedHypercertsList unclaimedHypercerts={data} />;
-};
+  return <UnclaimedHypercertsList unclaimedHypercerts={[...data]} />;
```

Comment:
What's the reason for ...spreading the results here?

Reply:
The action returns a readonly object and spreading creates a now non-readonly array

Reply:
This might be too nitpicky but do we know why/where it's marked as readonly? Because this just created an new array with references to the same objects as in the original `data`, which end up being mutable (which we shouldn't do anyways).

Reply:
Because that's what the `readFragment` method returns. It ensures typesafety and data consistency imho, we can view our Graph as the SoT and mutating API responses should be explicit. At least, that's how I interpret it.

The cleaner option could be to expect an `readonly` array in the component. Which I think is valid but we should apply in more places then

Reply:
Ok, I see now that the reason this case is different is because we are passing the entire list at once instead of iterating.